### PR TITLE
Add XSDAttribute format

### PR DESCRIPTION
**Pull request by adube**
This PR introduces the `ngeo.format.XSDAttribute` format, which allows the reading of attribute definitions in XSD and convert them into a list of attribute definitions, i.e. `Array.<ngeox.Attribute>`.

This PR introduces an example which is at a very "early" state. In the up-coming PRs, it will be updated to use the `ngeo-attributes` directive.  Don't mind it too much for now.

## Todo

 * [x] review 
 * [x] apply reviewer's comments / corrections

## Live example

 * http://adube.github.io/ngeo/xsd-format/examples/attributes.html

### CODE REVIEW

**Comment by adube** [comment]
@fredj Ready for review.

**Comment by adube** [comment]
Will do.

**Comment by adube** [comment]
I left the one in the constructor, though.

**Comment by fredj** [comment]
I'm now not so sure about how to determine if an attribute is required or not (`nillable` vs `minOccurs`) ...

After reading some random stackoverflow posts, I think that we should use `minOccurs > 0` for `required`.

@sbrunner what do you think ? Do we have some use cases ?

File: src/ol-ext/format/xsdattribute.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  var nillable = node.getAttribute('nillable');
+  var required = !(nillable === true || nillable === 'true');
+
+  var attribute = {
+    name: name,
+    required: required
+  };
+
+  var type = node.getAttribute('type');
+  if (type) {
```
Comment (fredj) [inline src/ol-ext/format/xsdattribute.js]: The type is required (in `ngeox.Attribute`); please remove this test and add a `goog.asserts.assert` just like you did for `name`

Comment (adube) [inline src/ol-ext/format/xsdattribute.js]: Aye aye, I'm on it.

File: src/ol-ext/format/xsdattribute.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      var choices = [];
+      for (var i = 0, ii = enumerations.length; i < ii; i++) {
+        choices.push(enumerations[i].getAttribute('value'));
+      }
+      attribute.choices = choices;
+    } else {
+      return null;
+    }
+  }
+
```
Comment (fredj) [inline src/ol-ext/format/xsdattribute.js]: please add `goog.asserts.assert(attribute.type);` here.

File: src/ol-ext/format/xsdattribute.js (hunk 1 above)
Comment (adube) [inline src/ol-ext/format/xsdattribute.js]: No wait, it's not always defined.  Yeah, you saw it too...

**Comment by sbrunner** [comment]
+1

**Comment by fredj** [comment]
@sbrunner do you know any project when an attribute is required? to know if we must use `nillable` or `minOccurs`

**Comment by sbrunner** [comment]
I think that we actually don't manage that...

**Comment by fredj** [comment]
ok, let's keep `nillable`, we can change it later

**Pull request merged**